### PR TITLE
Add sending method option in plugin config

### PR DESCRIPTION
**Pull request by gabriela-lubkowska**
- Added option in plugin config to choose sending method
- Added yellow background to orders with parcel locker as sending method while ordering courier
- Added description of yellow background in polish and english

### CODE REVIEW

File: src/Resources/app/administration/src/snippet/en-GB.json [hunk 1]
```diff
@@ -58,6 +58,8 @@
         "redBackgroundDescription": "The order(s) does not have a shipping method equal to InPost or the order(s) don't have created InPost package",
         "greenBackgroundTitle": "Green background",
         "greenBackgroundDescription": "A courier for this package(s) has already been ordered",
+        "yellowBackgroundTitle": "Yellow background",
+        "yellowBackgroundDescription": "The order(s) have a shipping method equal to InPost, but the package has selected parcel locker as sending method",
```
Comment (jakub-groncki) [inline src/Resources/app/administration/src/snippet/en-GB.json:62]: This and the one above about red background should be rephrased to something along the lines of:

The order's shipping method is InPost but the package's sending method is set to parcel locker.

Comment (gabriela-lubkowska) [reply src/Resources/app/administration/src/snippet/en-GB.json]: Changed in next commit

File: src/Migration/Migration1669886058OrderInPostExtension.php [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+{
+    public function getCreationTimestamp(): int
+    {
+        return 1669886058;
+    }
+
+    public function update(Connection $connection): void
+    {
+        $connection->executeStatement('ALTER TABLE `bitbag_inpost_point_order_extension`
+    ADD COLUMN `sending_method` VARCHAR(20) NULL');
```
[the remark this answers is not in the document]
Comment (gabriela-lubkowska) [reply src/Migration/Migration1669886058OrderInPostExtension.php]: Changed in next commit

File: src/Factory/Package/PackagePayloadFactory.php [hunk 3]
```diff
@@ -59,6 +69,18 @@ public function create(OrderEntity $order, Context $context): array
             ],
         ];
 
+        if (WebClientInterface::SENDING_METHOD_DISPATCH_ORDER == $sendingMethod) {
```
[the remark this answers is not in the document]
Comment (gabriela-lubkowska) [reply src/Factory/Package/PackagePayloadFactory.php]: I don't know if i understood it correctly, please check changes made in next commit

File: src/Extension/Content/Order/OrderInPostExtensionDefinition.php [hunk 4]
```diff
@@ -38,6 +38,7 @@ protected function defineFields(): FieldCollection
             new IntField('package_id', 'packageId'),
             new FkField('order_id', 'orderId', OrderDefinition::class),
             new OneToOneAssociationField('order', 'order_id', 'id', OrderDefinition::class, false),
+            new StringField('sending_method', 'sendingMethod', 20),
```
[the remark this answers is not in the document]
Comment (gabriela-lubkowska) [reply src/Extension/Content/Order/OrderInPostExtensionDefinition.php]: Changed in next commit

File: src/Controller/Api/PackageController.php [hunk 5]
```diff
@@ -76,12 +81,17 @@ public function __construct(
...
+    public function create(
+        string $orderId,
+        Context $context,
+        ?string $salesChannelId = null
+    ): JsonResponse {
         $order = $this->orderFinder->getWithAssociations($orderId, $context);
 
         $package = $this->packageApiService->createPackage($order, $context);
 
+        $sendingMethod = $this->inPostConfigService->getInPostApiConfig($salesChannelId)->getSendingMethod();
```
[the remark this answers is not in the document]
Comment (gabriela-lubkowska) [reply src/Controller/Api/PackageController.php:98]: Changed in next commit

File: src/Controller/Api/PackageController.php [hunk 6]
```diff
@@ -76,12 +81,17 @@ public function __construct(
      * )
      * @Route("/api/_action/bitbag-inpost-plugin/package/{orderId}", name="api.action.bitbag_inpost_plugin.package", methods={"POST"})
      */
-    public function create(string $orderId, Context $context): JsonResponse
-    {
+    public function create(
+        string $orderId,
+        Context $context,
+        ?string $salesChannelId = null
```
[the remark this answers is not in the document]
Comment (gabriela-lubkowska) [reply src/Controller/Api/PackageController.php:94]: Changed in next commit

File: src/Controller/Api/OrderCourierController.php [hunk 7]
```diff
@@ -209,6 +216,7 @@ private function getPackageAndOrderIds(array $ordersIds, Context $context): arra
                 ShippingMethodPayloadFactoryInterface::SHIPPING_KEY === $technicalName &&
                 null !== $inPostExtension &&
                 null !== $inPostExtension->get('packageId') &&
+                WebClientInterface::SENDING_METHOD_PARCEL_LOCKER !== $inPostExtension->get('sendingMethod') &&
```
[the remark this answers is not in the document]
Comment (gabriela-lubkowska) [reply src/Controller/Api/OrderCourierController.php:216]: Changed in next commit

File: src/Factory/Package/PackagePayloadFactory.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (jakub-groncki) [reply src/Factory/Package/PackagePayloadFactory.php]: Let's revert these changes but use `switch`, it'll be cleaner.

File: src/Factory/Package/PackagePayloadFactory.php [hunk 8]
```diff
@@ -77,4 +86,21 @@ private function addInsurance(array $data, OrderEntity $order): array
 
         return $data;
     }
+
+    private function checkSendingMethod(array $data, ?string $salesChannelId): array
+    {
+        $sendingMethod = $this->inPostConfigService->getInPostApiConfig($salesChannelId)->getSendingMethod();
```
Comment (jakub-groncki) [inline src/Factory/Package/PackagePayloadFactory.php:92]: Here in this method use `switch`, it'll be cleaner.

**Review by jakub-groncki — CHANGES_REQUESTED** [review]

**Review by jakub-groncki — APPROVED** [review]

**Pull request merged**